uniprotkb-rest/src/main/java/org/uniprot/api/uniprotkb/controller/UniProtKBController.java

```diff
@@ -218,6 +227,30 @@ public ResponseEntity<MessageConverterContext<UniProtKBEntry>> getByAccession(
         }
     }
 
+    private ResponseEntity<MessageConverterContext<UniProtKBEntry>> redirectToEntryVersion(
+            String accessionOrId, HttpServletRequest request) {
+        String version = null;
+        if (request.getParameter("version").equals("last")) {
+            try {
+                String entryVersionHistoryURI = "https://rest.uniprot.org/unisave/" + accessionOrId;
```

Comment:
Can you please use the logic bellow (in the method redirectToUniSave) to get UniSave path, extract a method and reuse in both places

Reply:
The redirectToUniSave method constructs the unisave URI and redirects the request to it. I think it's better to keep the logic for getting entry version separate from the unisave URI construction & redirection logic. What do you think?

The purpose of redirectToEntryVersion is to get the entry version and pass that version to unisave. Incase of "last" version we are calling unisave history endpoint to fetch all entry versions for the accession. Get the latest/last version and pass it to redirectToUniSave method.

Reply:
The reason that we implemented that way was because this code can also be executed with another URL, for example, when we create our dev environment, we will have a different URL for it, and it will probably be something like https://wwwdev.ebi.ac.uk/uniprot/api/unisave/
If we keep hardcoded rest.uniprot.org, it will be inconsistent in our dev environment and not redirecting to the correct unisave.